ci/docker/build.Dockerfile

```diff
@@ -94,13 +94,18 @@ RUN spack repo add --scope site /user_repo
 # Set this to a spack.yaml file which contains a spec
 # e.g. --build-arg SPACK_ENVIRONMENT=ci/spack/my-env.yaml
 ARG SPACK_ENVIRONMENT
+ARG COMMON_SPACK_ENVIRONMENT
 # Build dependencies
 # 1. Create a spack environment named `ci` from the input spack.yaml file
 COPY $SPACK_ENVIRONMENT /spack_environment/spack.yaml
 RUN spack env create --without-view ci /spack_environment/spack.yaml
-# 2. Set the C++ standard
+# 2. Copy the common environment configuration into the environment directory
+# Note that the destination path is hardcoded because we cannot dynamically get
+# the destination path. This will fail if the environment path ever changes.
+COPY $COMMON_SPACK_ENVIRONMENT /opt/spack/var/spack/environments/ci
```

Comment:
This is a bit ugly with the hardcoded path, but my Dockerfile knowledge is not enough to say if there are better alternatives to this. We can get the path with `spack location -e ci` but we can't use that in the `COPY` directive directly. I don't know if there are some roundabout ways of using it there.

At least it seems like spack will complain loudly if it can't include `common.yaml` so if this ever changes we would notice, but I'm open to better suggestions here.